File: rotate-image/rotate-image.py
class Solution:
    def rotate(self, m: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        lenm = len(m)
        # algorithm to transpose square matrix in place, swap all elements in left of diagonal with right of diagonal
        for i in range(lenm):
            for j in range(0,i):
                temp = m[i][j]
                m[i][j] = m[j][i]
                m[j][i] = temp

        # reverse every row in matrix
        for i in range(lenm):
            for j in range(int(lenm/2)):
                temp = m[i][j]
                m[i][j] = m[i][lenm-1-j]
                m[i][lenm-1-j] = temp
                

File: rotate-image/rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        for(int i=0; i<matrix.length; i++) {
            for(int j=i+1; j<matrix[0].length; j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }
        
        for(int j=0; j<matrix.length; j++) {
            var row = matrix[j];
            int[] revRow = new int[row.length];
            for(int i=0; i<row.length; i++) {
                revRow[row.length-1-i] = row[i];
            }
            matrix[j] = revRow;
        }
    }
}